src/Stratis.Bitcoin.IntegrationTests/BlockStore/ReOrgRegularlySpecification_Steps.cs

```diff
@@ -0,0 +1,156 @@
+
+            this.nodes[NodeB].FullNode.NodeService<WalletController>().SendTransaction(new SendTransactionRequest(this.shorterChainTransaction.ToHex()));
+        }
+
+        private HdAddress GetSecondUnusedAddressToAvoidClashWithMiningAddress(CoreNode node)
+        {
+            return node.FullNode.WalletManager()
+                .GetUnusedAddresses(new WalletAccountReference(WalletZero, AccountZero), 2)
+                .Skip(1).First();
+        }
+
+        private void nodeC_mines_this_block()
+        {
+            this.sharedSteps.MineBlocks(1, this.nodes[NodeC], AccountZero, WalletZero, WalletPassword, this.shortChainTransactionFee.Satoshi);
+        }
+
+        private void nodeD_confirms_it_ensures_tx_present()
+        {
+            this.sharedSteps.WaitForBlockStoreToSync(this.nodes[NodeB], this.nodes[NodeC], this.nodes[NodeD]);
+
+            var transaction = this.nodes[NodeD].FullNode.BlockStoreManager().BlockRepository.GetTrxAsync(this.shorterChainTransaction.GetHash()).Result;
+            transaction.Should().NotBeNull();
+            transaction.GetHash().Should().Be(this.shorterChainTransaction.GetHash());
+        }
+
+        private void selfish_node_reconnects_and_broadcasts()
+        {
+            this.nodes[SelfishMiner].CreateRPCClient().AddNode(this.nodes[NodeB].Endpoint);
+            this.sharedSteps.WaitForBlockStoreToSync(this.nodes[SelfishMiner], this.nodes[NodeB], this.nodes[NodeC], this.nodes[NodeD]);
+        }
+         
+        private void other_nodes_reorg_to_longest_chain()
+        {
+            TestHelper.WaitLoop(() => this.nodes[NodeB].FullNode.Chain.Height == this.selfishBlockHeight);
+            this.nodes[NodeB].FullNode.Chain.Height.Should().Be(this.selfishBlockHeight);
+            TestHelper.WaitLoop(() => this.nodes[NodeC].FullNode.Chain.Height == this.selfishBlockHeight);
+            this.nodes[NodeC].FullNode.Chain.Height.Should().Be(this.selfishBlockHeight);
+            TestHelper.WaitLoop(() => this.nodes[NodeD].FullNode.Chain.Height == this.selfishBlockHeight);
+            this.nodes[NodeD].FullNode.Chain.Height.Should().Be(this.selfishBlockHeight);
+        }
+
+        private void transaction_from_shorter_chain_is_missing()
+        {
+            this.nodes[NodeB].FullNode.BlockStoreManager().BlockRepository.GetTrxAsync(this.shorterChainTransaction.GetHash()).Result
+                .Should().BeNull("longest chain comes from selfish miner and shouldn't contain the transaction made on the chain with the other 3 nodes.");
+        }
+
+        private void transaction_is_not_returned_to_the_mem_pool()
+        {
+            this.nodes[NodeD].CreateRPCClient().GetRawMempool()
+                .Should().NotContain(x => x == this.shorterChainTransaction.GetHash(), "it is not implemented yet.");
+        }
+
+        private void mining_continues_to_maturity_to_allow_spend()
+        {
+            var coinbaseMaturity = (int)this.nodes[NodeB].FullNode
```

Comment:
Any significance of picking NodeB - over the other nodes - to get the coinbaseMaturity ?

Reply:
nope